packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -326,33 +333,49 @@ abstract class LocalHistoryRoute<T> extends Route<T> {
   bool get willHandlePopInternally {
     return _localHistory != null && _localHistory.isNotEmpty;
   }
+
+  /// Called whenever the internal state of the route has changed.
+  ///
+  /// This should be called whenever [willHandlePopInternally] and [didPop]
+  /// might change the value they return. It is used by [ModalRoute], for
+  /// example, to report the new information via its inherited widget to any
+  /// children of the route.
+  @protected
+  @mustCallSuper
+  void changedInternalState() { }
```

Comment:
I didn't really want it to sound the same as the others since the others are called by the Navigator